packages/kbn-esql-validation-autocomplete/src/validation/validation.ts

```diff
@@ -75,12 +77,33 @@ import {
 function validateFunctionLiteralArg(
   astFunction: ESQLFunction,
   actualArg: ESQLAstItem,
-  argDef: SignatureArgType,
+  argDef: FunctionArg,
   references: ReferenceMaps,
   parentCommand: string
 ) {
   const messages: ESQLMessage[] = [];
   if (isLiteralItem(actualArg)) {
+    if (
+      // currently only supporting literalOptions with string literals
+      actualArg.literalType === 'string' &&
+      argDef.literalOptions &&
+      !argDef.literalOptions
+        .map((option) => option.toLowerCase())
+        .includes(unwrapStringLiteralQuotes(actualArg.value).toLowerCase())
```

Comment:
I would probably make this go into the omniscent `isEqualType` as:

```ts
// currently only supporting literalOptions with string literals
      actualArg.literalType === 'string' &&
      argDef.literalOptions &&
      !isEqualType(actualArg, argDef, references, parentCommand)
```

then the `isEqualType` signature will change as:

```ts
export function isEqualType(
  item: ESQLSingleAstItem,
  argDef: SignatureArgType | FunctionArg,
  references: ReferenceMaps,
  parentCommand?: string,
  nameHit?: string
) {
  const argType = 'innerType' in argDef && argDef.innerType ? argDef.innerType : argDef.type;
  if (argType === 'any') {
    return true;
  }
  if (item.type === 'literal') {
    return compareLiteralType(
      argType,
      item,
      'literalOptions' in argDef ? argDef.literalOptions : undefined
    );
  }
...
```

and the `compareLiteralType` function becomes:
```ts
function compareLiteralType(argTypes: string, item: ESQLLiteral, possibleOptions?: string[]) {
  if (item.literalType !== 'string') {
    return argTypes === item.literalType;
  }
  if (argTypes === 'chrono_literal') {
    return chronoLiterals.some(({ name }) => name === item.text);
  }
  if (possibleOptions) {
    return possibleOptions.includes(unwrapStringLiteralQuotes(item.text.toLowerCase()));
  }
  return argTypes === item.literalType;
}
```

(I think the assumption that the `literalOptions` are already lower case can be made).

Reply:
I started down this path but I'm not sure `isEqualType` is the best fit. We want a different message if the type is correct but it doesn't match one of the `literalOptions`. All `isEqualType` returns is `true` or `false`. 

If I were to incorporate this check into `isEqualType`, I would end up reporting that the type is incorrect when, in fact, it just doesn't match one of the supported options.